__tests__/transactions/ViewMosaicDefinitionTransaction.spec.ts

```diff
@@ -0,0 +1,119 @@
+import { ViewMosaicDefinitionTransaction } from '@/core/transactions/ViewMosaicDefinitionTransaction';
+import { OfflineRentalFees } from '@/services/offline/MockModels';
+
+describe('transactions/ViewMosaicSupplyRevocationTransaction', () => {
+    const signerPublicAccount = Account.generateNewAccount(NetworkType.TEST_NET).publicAccount;
+    const store = createStore({});
+    const epochAdjustment = 1573430400;
+    const blockGenerationTargetTime = 30;
+    const randomNonce = MosaicNonce.createRandom();
+    const mosaicId = MosaicId.createFromNonce(randomNonce, signerPublicAccount.address);
+    const defaultMosaicFlags = MosaicFlags.create(false, false, false, false);
+
+    beforeEach(() => {
+        const networkConfig = new NetworkConfigurationModel();
+        const rentalFees = OfflineRentalFees;
+        Object.assign(networkConfig, { epochAdjustment: epochAdjustment, blockGenerationTargetTime: blockGenerationTargetTime });
+        store.getters['network/networkConfiguration'] = networkConfig;
+        store.getters['network/rentalFeeEstimation'] = rentalFees;
+    });
+
+    const buildTransaction = (
+        mosaicFlags: MosaicFlags = defaultMosaicFlags,
+        divisibility: number = 6,
+        duration: UInt64 = UInt64.fromUint(0),
+    ) => {
+        return MosaicDefinitionTransaction.create(
+            Deadline.create(epochAdjustment),
+            randomNonce,
+            mosaicId,
+            mosaicFlags,
+            divisibility,
+            duration,
+            NetworkType.TEST_NET,
+            UInt64.fromUint(1_000_000),
+        );
+    };
+
+    const assertCommonFields = (view: ViewMosaicDefinitionTransaction) => {
+        expect(view).toBeDefined();
+        expect(view.transaction).toBeDefined();
+        expect(view.detailItems.length).toBe(8);
+    };
+
+    const testPopulateDataForMosaicDefinitionWithDifferentMosaicFlagsView = (mosaicFlags: MosaicFlags) => {
+        // Arrange:
+        const transaction = buildTransaction(mosaicFlags);
+
+        // Act:
+        const view = new ViewMosaicDefinitionTransaction(store, transaction);
+
+        // Assert:
+        assertCommonFields(view);
+        expect(view.detailItems).toContainEqual({ key: 'table_header_transferable', value: mosaicFlags.transferable });
+        expect(view.detailItems).toContainEqual({ key: 'table_header_supply_mutable', value: mosaicFlags.supplyMutable });
+        expect(view.detailItems).toContainEqual({ key: 'table_header_restrictable', value: mosaicFlags.restrictable });
+        expect(view.detailItems).toContainEqual({ key: 'table_header_revokable', value: mosaicFlags.revokable });
+        expect(view.detailItems).toContainEqual({ key: 'table_header_divisibility', value: '6' });
+        expect(view.detailItems).toContainEqual({ key: 'estimated_rental_fee', isMosaic: true, value: { amount: 500000, color: 'red' } });
+        expect(view.detailItems).toContainEqual({ key: 'mosaic_id', value: mosaicId.toHex() });
```

Comment:
you may consider using `toEqual(object)` instead of `toContainEqual`.

Reply:
`duration` property is missing.

Reply:
I changed tests a bit. I have one case where I'm testing all entries and another for specific cases. Please have a look.